analytics/reports/orders.md

```diff
@@ -0,0 +1,567 @@
+          oh.ENTRY_DATE,
+          oh.ORDER_DATE "Created At",
+          max(os.status_datetime) "Approved At"
+   from order_header oh
+   join order_status os on os.ORDER_ID = oh.ORDER_ID
+   and os.status_id = "ORDER_APPROVED"
+   where os.status_datetime
+   group by oh.ORDER_ID
+   order by oh.ORDER_ID desc) AS virtual_table
+WHERE `Created At` >= STR_TO_DATE('2024-04-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `Created At` < STR_TO_DATE('2024-05-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `ENTRY_DATE` >= STR_TO_DATE('2024-04-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+  AND `ENTRY_DATE` < STR_TO_DATE('2024-05-08 00:00:00.000000', '%Y-%m-%d %H:%i:%s.%f')
+GROUP BY DATE(`ENTRY_DATE`)
+ORDER BY `Average approval duration in minutes` DESC
+LIMIT 10000;
+```
+
+</details>
+
+**Data Selection:** The SQL query begins by selecting specific data from two tables: `order_header` and `order_status`. These tables likely contain information about orders and their statuses.
+
+**Defining Criteria:** The query filters the data to include only orders with a specific status, in this case, orders that are approved. This is achieved by specifying the condition `os.status_id = 'ORDER_APPROVED'`.
+
+**Joining Relevant Tables:** To combine relevant information from both tables, the query joins `order_header` and `order_status` using the common field `ORDER_ID`.
+
+**Selecting Necessary Information:** From the joined tables, the query selects columns such as order ID, Shopify order ID, approval duration (calculated in seconds), creation date (`ORDER_DATE`), and approval date (`status_datetime`).
+
+**Grouping Data:** The query groups the selected data by order ID. This means that all entries with the same order ID are grouped together, allowing for aggregation and analysis of approval durations for each order. The query groups the results by the date part of the creation date (`ENTRY_DATE`), allowing for a summary of average approval durations and order volumes for each day.
+
+**Calculating Average Approval Duration:** Within each group, the query calculates the average approval duration in minutes (`AVG(duration in sec) / 60`).
+
+**Counting Distinct Orders:** Additionally, the query counts the number of distinct order IDs to determine the volume of orders for each day.
+
+## Missing Order Attribute Report
+
+The Missing Order Attribute Report is a vital tool for tracking order synchronization. By monitoring the presence of essential attributes, it identifies orders lacking crucial information, ensuring a seamless synchronization process. This report enables proactive resolution of discrepancies, preventing any orders from failing to synchronize effectively. In the NetSuite context, the report ensures accurate order synchronization by verifying essential attributes. By highlighting orders lacking these attributes, it prevents synchronization issues, providing assurance that orders seamlessly integrate with NetSuite.
```

Comment:
not tool but report
as mentioned before, give a brief description of what are order attributes